pkg/virt-handler/vm_test.go

```diff
@@ -1701,6 +1700,181 @@ var _ = Describe("VirtualMachineInstance", func() {
 	})
 })
 
+var _ = Describe("DomainNotifyServerRestarts", func() {
+	Context("should establish a notify server pipe", func() {
+		var shareDir string
+		var err error
+		var serverStopChan chan struct{}
+		var serverIsStoppedChan chan struct{}
+		var stoppedServer bool
+		var domainPipeStopChan chan struct{}
+		var stoppedPipe bool
+		var eventChan chan watch.Event
+		var client *notifyclient.Notifier
+		var recorder *record.FakeRecorder
+		var vmiStore cache.Store
+
+		BeforeEach(func() {
+			serverStopChan = make(chan struct{})
+			domainPipeStopChan = make(chan struct{})
+			serverIsStoppedChan = make(chan struct{})
+			eventChan = make(chan watch.Event, 100)
+			stoppedServer = false
+			stoppedPipe = false
+			shareDir, err = ioutil.TempDir("", "kubevirt-share")
+			Expect(err).ToNot(HaveOccurred())
+
+			recorder = record.NewFakeRecorder(10)
+			vmiInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstance{})
+			vmiStore = vmiInformer.GetStore()
+
+			go func(serverIsStoppedChan chan struct{}) {
+				notifyserver.RunServer(shareDir, serverStopChan, eventChan, recorder, vmiStore)
+				close(serverIsStoppedChan)
+			}(serverIsStoppedChan)
+
+			time.Sleep(3)
+		})
+
+		AfterEach(func() {
+			if stoppedServer == false {
+				close(serverStopChan)
+			}
+			if stoppedPipe == false {
+				close(domainPipeStopChan)
+			}
+			client.Close()
+			os.RemoveAll(shareDir)
+		})
+
+		It("should get notify events", func() {
+			vmi := v1.NewMinimalVMI("fake-vmi")
+			vmi.UID = "4321"
+			vmiStore.Add(vmi)
+
+			eventType := "Normal"
+			eventReason := "fooReason"
+			eventMessage := "barMessage"
+
+			pipePath := filepath.Join(shareDir, "client_path", "domain-notify-pipe.sock")
```

Comment:
This hardcoded pipe name does not match the one that handleDomainNotifyPipe is looking for. Is that on purpose?

Reply:
yep, the way this was structured to make it unit testable is a bit confusing. 

we create a pipe listener, then pass that listener in to handleDomainNotifyPipe() which plugs the incoming connections from the listener to the backend server.